sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/api/Azure.ResourceManager.RedisEnterprise.net8.0.cs

```diff
@@ -432,6 +489,109 @@ protected virtual void JsonModelWriteCore(System.Text.Json.Utf8JsonWriter writer
+        public static bool operator !=(Azure.ResourceManager.RedisEnterprise.Models.MaintenanceDayOfWeek left, Azure.ResourceManager.RedisEnterprise.Models.MaintenanceDayOfWeek right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class MaintenanceWindow : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>
+    {
+        public MaintenanceWindow(Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowType windowType, System.TimeSpan duration, int startHourUtc, Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule schedule) { }
+        public System.TimeSpan Duration { get { throw null; } set { } }
+        public Azure.ResourceManager.RedisEnterprise.Models.MaintenanceDayOfWeek? ScheduleDayOfWeek { get { throw null; } set { } }
+        public int StartHourUtc { get { throw null; } set { } }
+        public Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowType WindowType { get { throw null; } set { } }
+        protected virtual void JsonModelWriteCore(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options) { }
+        Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.Create(ref System.Text.Json.Utf8JsonReader reader, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        void System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.Write(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options) { }
+        Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.Create(System.BinaryData data, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindow>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+    }
+    public partial class MaintenanceWindowSchedule : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.RedisEnterprise.Models.MaintenanceWindowSchedule>
```

Comment:
name too generic

Reply:
Renamed to `ClusterCustomMaintenanceWindowSchedule`